lib/usd/utils/util.cpp

```diff
@@ -142,26 +142,26 @@ printCompositionQuery(const UsdPrim& prim, std::ostream& os)
 }
 
 bool
-updateInternalReferences(const UsdPrim& oldPrim, const UsdPrim& newPrim)
+updateInternalReferencesPath(const UsdPrim& oldPrim, const SdfPath& newPath)
 {
-    for (const auto& p : newPrim.GetStage()->TraverseAll()) {
-
-        // check is this prim has a spec
+    SdfChangeBlock changeBlock;
```

Comment:
Added SdfChangeBlock to make the loop more performant and delay sending notification until after we exist this function.

Based on what we discussed internally the other day, having SdfChangeBlock on the stack while making calls into the Usd API's could be troublesome but I didn't see any issue in this case.